### PR TITLE
move to setuptools

### PR DESCRIPTION
**Pull request by rardiol**
Removed the old singularity.py and release submodules and moved to modern python packaging with [setuptools](https://packaging.python.org/). This is my first foray into python packaging, so please correct me if I did anything wrong.

Still should be done :
- Update README.txt
- Bump version number to 0.30-4 (note that setuptools doesn't like letter sufixes) and make a release
- Submit to PyPI. Actually, it's already there, abandoned: https://pypi.python.org/pypi/Endgame-Singularity, so we will probably need @evilmrhenry account.
- Possibly use this: https://pythonhosted.org/setuptools/pkg_resources.html#resourcemanager-api.

### CODE REVIEW

File: release/README.txt [hunk 1]
```diff
@@ -1,74 +0,0 @@
...
-    https://github.com/singularity/singularity-windows/archive/v0.30c.zip
-
-Will trigger the download of a file named `singularity-windows-0.30c.zip`.
-
-If you are a developer and want to create and publish a new release, first
-fetch and update its submodules. From the singularity root directory, run:
-
-git submodule update --init release/windows
-
----
```
Comment (evilmrhenry) [inline release/README.txt]: Those "release steps" should still be present somewhere. They likely need updates, but are around 90% accurate and a useful checklist for releases.

**Comment by evilmrhenry** [comment]
It's saying the branch has conflicts.

**Comment by MestreLion** [comment]
Also, this is a very wide change, that accomplishes much more than just updating `setup.py`, so perhaps it's best if split into multiple commits? Specially the (welcome) `code` -> `singularity` dir renaming should be in its own commit.

**Comment by rardiol** [comment]
Better now?

**Comment by evilmrhenry** [comment]
I'm ok with it, but I want some other opinions.

**Comment by Quix0r** [comment]
MANIFEST.in misses a new-line character at the end of the file. Some editors may be confused or event crash if that line is not there.

**Comment by MestreLion** [comment]
Care to rebase this patch so we can merge it?

**Comment by rardiol** [comment]
rebased.

**Comment by Xenega** [comment]
Can you rebase it ?

**Comment by rardiol** [comment]
I tried and it seemed like a nontrivial rebase with some code moves, and to be honest I did this so long ago I don't really remember the details of what I did, will finish the rebase later.

**Comment by Xenega** [comment]
Yes, I am doing a lot of change. Thanks.

**Comment by Xenega** [comment]
Since a lot of refactoring and change will be come next, if you can't do it in one time, I suggest you to wait I freeze before release.

**Comment by nthykier** [comment]
Fixed in aba8ba607f286b331051caae32ebfa9c77297f6e based on commits by @rardiol.

@rardiol Thanks for your initial work on this task!

**Pull request closed without merging**